### PR TITLE
migrate database if app.ini found

### PR DESCRIPTION
**Pull request by pciavald**
Hi, 

For my installation of gitea, I needed unattended initial config. @techknowlogick told me on Discord that it could be done using then new migrate command, `gitea migrate -c /data/gitea/conf/app.ini`. 

I've added in the Docker entrypoint:
- check if `/data/gitea/conf/app.ini` exists
- change mode and owner
- migrate database

It works correctly on my machine, in my docker-compose the data volume is mapped with only the config file and the platform auto-inits. However I can see in the logs this error which does not affect behaviour:

```
migrate: /data/gitea/conf/app.ini: Permission denied
```

### CODE REVIEW

File: docker/usr/bin/entrypoint
Comment (pciavald) [reply docker/usr/bin/entrypoint:38]: `${USER_UID}`, `${USER_GID}`, `$UID` and `$GID` are undefined, i'll use `$USER:$USER` is that ok ?

Comment (zeripath) [reply docker/usr/bin/entrypoint:38]: The original entry point script just checks for the variables and if they're there does something. That's why they're undefined.

You may have to adjust it to define the variables if they're not there or empty.

```bash
if [ -z "${USER_GID}" ]; then
  USER_GID="`id -g ${USER}`"
fi
```

And similarly for UID. Otherwise I'd be suspicious that the chown could choose the wrong UID and GID in cases were they were set explicitly - but I'd happily defer to more expert opinion on this.

Comment (techknowlogick) [reply docker/usr/bin/entrypoint:38]: Ah yes. You are right @pciavald, I missed that they were undefined. I think that @zeripath's approach of setting a default for the values is a good one.

File: docker/usr/bin/entrypoint
Comment (zeripath) [inline docker/usr/bin/entrypoint]: Heya!

Just remembering that this is still stuck in the queue and I think it's probably a good idea to have this.

To get my LGTM you need to add to the top of file (or as appropriate):

```bash
if [ -z "${USER_GID}" ]; then
  USER_GID="`id -g ${USER}`"
fi

if [ -z "${USER_UID}" ]; then
  USER_UID="`id -u ${USER}`"
fi
```

and change the chown to:

```bash
chown -R ${USER_UID}:${USER_GID} /data/git /data/gitea
```

**Review by zeripath — CHANGES_REQUESTED** [review]

File: docker/usr/bin/entrypoint
Comment (techknowlogick) [reply docker/usr/bin/entrypoint]: @zeripath I've just pushed these changes.

**Review by techknowlogick — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

File: docker/usr/bin/entrypoint
Comment (apricote) [inline docker/usr/bin/entrypoint:35]: This will allow anyone with access to the server to read the `app.ini`, exposing any credentials (database, metrics, smtp).

**Review by apricote — COMMENTED** [review]
I think there is a small security issue caused by this PR

File: docker/usr/bin/entrypoint
Comment (zeripath) [reply docker/usr/bin/entrypoint:35]: Hmm... I guess you're right. I guess this should be `600`. In some ways this is not so bad because this for docker - you shouldn't have other users on it in any case, that's not the docker way.

Comment (apricote) [reply docker/usr/bin/entrypoint:35]: Is there any reason to change the file permissions?

Comment (zeripath) [reply docker/usr/bin/entrypoint:35]: Possibly not. If you've not set the `INSTALL_LOCK` setting in your `app.ini` and you've got the wrong mode then you won't be able to use `/install`. However I guess we shouldn't be being preventing people from shooting themselves in the foot like that. Pop a PR on if you're able otherwise I can when I'm next at a dev box. @techknowlogick do you agree that perhaps we shouldn't changing the mode here?

Comment (zeripath) [reply docker/usr/bin/entrypoint:35]: @pciavald What was your reasoning for changing the mode?

**Comment by zeripath** [comment]
@techknowlogick I think unfortunately there may still be problems with this.

The correct place for automigration is likely within the gitea script within the  s6 folder. That would require a little refactoring of that to make it work.

However a simple improvement is that we should change the migrate line to:

```
(cd /app/gitea; exec su-exec $USER /app/gitea/gitea migrate)
```

 as per my comment:

https://github.com/go-gitea/gitea/issues/5688#issuecomment-453876808

Then this will more closely match how docker starts Gitea so if people have used relative paths etc in their app.ini we're not affected by these.

I won't have a chance to write a pr for this for several days so if you can or get a friendly minion please do.

**Pull request merged**
